src/simpleweighteddigraph.jl

```diff
@@ -105,13 +117,39 @@ function add_edge!(g::SimpleWeightedDiGraph, e::SimpleWeightedGraphEdge)
     return true
 end
 
-function rem_edge!(g::SimpleWeightedDiGraph, e::SimpleWeightedGraphEdge)
-    has_edge(g, e) || return false
-    U = weighttype(g)
-    @inbounds g.weights[dst(e), src(e)] = zero(U)
+rem_edge!(g::SimpleWeightedDiGraph{T, U}, e::AbstractEdge) where {T<:Integer, U<:Real} =
+    rem_edge!(g, src(e), dst(e))
+
+function rem_edge!(g::SimpleWeightedDiGraph{T, U}, u::Integer, v::Integer) where {T<:Integer, U<:Real}
+    u ∈ vertices(g) && v ∈ vertices(g) || return false
+    w = g.weights
+    indx = _get_nz_index!(w, v, u)
+    indx == 0 && return false
+    @view(w.colptr[u+one(u):end]) .-= T(1)
+    deleteat!(w.rowval, indx)
+    deleteat!(w.nzval, indx)
+
     return true
 end
 
+@doc_str """
+    rem_vertex!(g::SimpleWeightedDiGraph, v)
+
+Remove the vertex `v` from graph `g`. Return false if removal fails
+(e.g., if vertex is not in the graph); true otherwise.
+
+### Implementation Notes
+This operation has to be performed carefully if one keeps external
+data structures indexed by edges or vertices in the graph, since
+internally the removal results in all vertices with indices greater than `v`
+being shifted down one.
+"""
+function rem_vertex!(g::SimpleWeightedDiGraph, v::Integer)
+    v in vertices(g) || return false
+    n = nv(g)
+    g.weights = g.weights[1:n .!= v, 1:n .!= v]
```

Comment:
Does this allocate?

Reply:
Probably not because if we do a view it will change the type of `g.weights`. That sucks though

Reply:
```suggestion
    all_except_v = (1:n) .!= v
    g.weights = g.weights[all_except_v, all_except_v]
```

Reply:
Honestly, the bottleneck here is clearly the allocation of the new weight matrix, and this optimization is probably already done by the compiler